frontend/utils/bridge.ts

```diff
@@ -14,72 +14,121 @@
-    resolveGoogleLogin: (data) => console.log("Google login data received:", data),
-    rejectGoogleLogin: (err) => console.error("Google login failed:", err),
-    requestUploadToGoogleDrive: (data) => window.ReactNativeWebView.postMessage(JSON.stringify({ topic: "upload_to_google_drive", data })),
-    resolveUploadToGoogleDrive: () => console.log("Google Drive upload successful"),
-    rejectUploadToGoogleDrive: (err) => console.error("Google Drive upload failed:", err),
-    requestRestoreGoogleDriveBackup: () => window.ReactNativeWebView.postMessage(JSON.stringify({ topic: "restore_google_drive_backup" })),
-    resolveRestoreGoogleDriveBackup: (data) => console.log("Google Drive backup restored:", data),
-    rejectRestoreGoogleDriveBackup: (err) => console.error("Google Drive restore failed:", err),
-    requestGoogleAuthState: () => window.ReactNativeWebView.postMessage(JSON.stringify({ topic: "check_google_auth_state" })),
-    resolveGoogleAuthState: (data) => console.log("Google Auth State:", data),
-    rejectGoogleAuthState: (err) => console.error("Google Auth State check failed:", err),
-    requestGoogleUserInfo: () => window.ReactNativeWebView.postMessage(JSON.stringify({ topic: "google_user_info" })),
-    resolveGoogleUserInfo: (data) => console.log("Google User Info:", data),
-    rejectGoogleUserInfo: (err) => console.error("Google User Info retrieval failed:", err)
+      
+      // Return a promise for new Promise-based usage
+      return new Promise((resolve, reject) => {
+        window.nativebridge._pendingPromises[requestId] = { resolve, reject };
+      });
+    },`);
+
+    /**
+     * Generate resolve method - resolves the corresponding promise
+     */
+    methods.push(`
+    ${resolveMethod}: (data, requestId) => {
+      console.log("${topic} resolved:", data);
+      if (window.nativebridge._pendingPromises && window.nativebridge._pendingPromises[requestId]) {
+        window.nativebridge._pendingPromises[requestId].resolve(data);
+        delete window.nativebridge._pendingPromises[requestId];
+      }
+    },`);
+
+    /**
+     * Generate reject method - rejects the corresponding promise
+     */
+    methods.push(`
+    ${rejectMethod}: (error, requestId) => {
+      console.error("${topic} failed:", error);
+      if (window.nativebridge._pendingPromises && window.nativebridge._pendingPromises[requestId]) {
+        window.nativebridge._pendingPromises[requestId].reject(error);
+        delete window.nativebridge._pendingPromises[requestId];
+      }
+    },`);
+
+    /**
+     * Generate helper methods for data persistence across page reloads
+     * Creates global variables and getter methods for storing resolved data
+     */
+
+    // Create a generic global variable for this topic
+    const globalVarName = `native${topic.split('_').map(word =>
+      word.charAt(0).toUpperCase() + word.slice(1)
+    ).join('')}`;
+
+    globalHelpers.push(`  window.${globalVarName} = null;`);
```

Comment:
Is the white space expected?
```suggestion
    globalHelpers.push(`window.${globalVarName} = null;`);
```